subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/SwiftProjectInitDescriptor.java

```diff
@@ -0,0 +1,132 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.buildinit.plugins.internal;
+
+import com.google.common.collect.Sets;
+import org.gradle.buildinit.plugins.internal.modifiers.BuildInitTestFramework;
+import org.gradle.buildinit.plugins.internal.modifiers.Language;
+import org.gradle.internal.os.OperatingSystem;
+import org.gradle.nativeplatform.platform.internal.DefaultNativePlatform;
+
+import java.util.Set;
+
+import static org.gradle.buildinit.plugins.internal.ModuleNameBuilder.toModuleName;
+
+public abstract class SwiftProjectInitDescriptor extends LanguageLibraryProjectInitDescriptor {
+    private final TemplateOperationFactory templateOperationFactory;
+
+    public SwiftProjectInitDescriptor(TemplateOperationFactory templateOperationFactory) {
+        this.templateOperationFactory = templateOperationFactory;
+    }
+
+    @Override
+    public Language getLanguage() {
+        return Language.SWIFT;
+    }
+
+    @Override
+    public void generate(InitSettings settings, BuildScriptBuilder buildScriptBuilder, TemplateFactory templateFactory) {
+        buildScriptBuilder
+            .fileComment("This generated file contains a sample Swift project to get you started.")
+            .fileComment("For more details take a look at the Building Swift applications and libraries chapter in the Gradle");
+        configureBuildScript(settings, buildScriptBuilder);
+
+        TemplateOperation sourceTemplate = sourceTemplateOperation(settings);
+        TemplateOperation testSourceTemplate = testTemplateOperation(settings);
+
+        if (OperatingSystem.current().isLinux() || OperatingSystem.current().isWindows()) {
+            TemplateOperation testEntryPointTemplate = testEntryPointTemplateOperation(settings);
+            templateFactory.whenNoSourcesAvailable(sourceTemplate, testSourceTemplate, testEntryPointTemplate).generate();
+        } else {
+            templateFactory.whenNoSourcesAvailable(sourceTemplate, testSourceTemplate).generate();
+        }
+    }
+
+    @Override
+    public Set<BuildInitTestFramework> getTestFrameworks() {
+        return Sets.newHashSet();
+    }
+
+    @Override
+    public BuildInitTestFramework getDefaultTestFramework() {
+        return null;
+    }
+
+    protected abstract TemplateOperation sourceTemplateOperation(InitSettings settings);
+
+    protected abstract TemplateOperation testTemplateOperation(InitSettings settings);
+
+    protected abstract TemplateOperation testEntryPointTemplateOperation(InitSettings settings);
+
+    protected void configureBuildScript(InitSettings settings, BuildScriptBuilder buildScriptBuilder) {
+    }
+
+    @Override
+    public boolean supportsPackage() {
```

Comment:
Can we support this for Swift?

Reply:
I don't think it's something Swift support. C++ have some sort of "package" (e.g. namespace) but we still return `false` here. It used to support `--package`. I don't think we need to support this now.